packages/test/src/test/tests.ts

```diff
@@ -9,417 +9,419 @@ import sm from "source-map-support"
 sm.install()
 const DATA: string = "./data/"
 
-describe("Repository tests", () => {
-    const t = new RepositoryClient()
-    let initialPartition: LionWebJsonChunk
-    let initialPartitionVersion: number = 0
-    let baseFullChunk: LionWebJsonChunk
-    let baseFullChunkVersion: number = 0
+const collection = [true, false]
 
-    before("create database", async function () {
-        const initResponse = await t.createDatabase()
-        if (initResponse.status !== HttpSuccessCodes.Ok) {
-            console.log("Cannot create database: " + JSON.stringify(initResponse.body))
-        } else {
-            console.log("database created: " + JSON.stringify(initResponse.body))
-        }
-    })
-    
-    beforeEach("a", async function () {
-        initialPartition = t.readModel(DATA + "Disk_A_partition.json") as LionWebJsonChunk
-        baseFullChunk = t.readModel(DATA + "Disk_A.json") as LionWebJsonChunk
-        const initResponse = await t.init()
-        if (initResponse.status !== HttpSuccessCodes.Ok) {
-            console.log("Cannot initialize database: " + JSON.stringify(initResponse.body))
-        } else {
-            console.log("initialized database: " + JSON.stringify(initResponse.body))
-        }
-        const partResult = await t.testCreatePartitions(initialPartition)
-        if (partResult.status !== HttpSuccessCodes.Ok) {
-            console.log("Cannot create initial partition: " + JSON.stringify(partResult.body))
-            console.log(JSON.stringify(initialPartition))
-        }
-        // console.log("CREATE PARTITIONS RESULT " + JSON.stringify(partResult))
-        initialPartitionVersion = Number.parseInt(partResult.body.messages.find(m => m.data["version"] !== undefined).data["version"])
-        const result = await t.testStore(baseFullChunk)
-        if (result.status !== HttpSuccessCodes.Ok) {
-            console.log("Cannot store initial chunk: " + JSON.stringify(result.body))
-        }
-        // console.log("STORE PARTITIONS RESULT " + JSON.stringify(result.body.messages))
-        const tmp2 = result.body.messages.find(m => m.data["version"] !== undefined).data["version"]
-        console.log("TMP 2 [" + tmp2 + "]")
-        baseFullChunkVersion = Number.parseInt(result.body.messages.find(m => m.data["version"] !== undefined).data["version"])
-        console.log("repoVersionAfterPartitionCreated " + initialPartitionVersion + "repoVersionAfterPartitionFilled " + baseFullChunkVersion)
-    })
+// Run all, tests with and without history
+collection.forEach(withoutHistory =>
```

Comment:
Nice! (`collection` could be inlined.)

Reply:
Strangely enough I get a TypeScript error message when I inline the collection: TS2538: Type false cannot be used as an index type

Reply:
Hmmm, weird...but also not really important :)